### PR TITLE
Add commands modelling and api response to server

### PR DESCRIPTION
**Pull request by AY-Chen**
Add model for commands(ie manual overrides to scheduling) and unit tests. 
Add basic server response and handling of POST and DELETE requests. 
MISSING: still need to force update for motor.

### CODE REVIEW

**Comment by AY-Chen** [comment]
I see the CI error. I forgot to update the file. Give me a minute

**Comment by AY-Chen** [comment]
> 
> 
> I see the CI error. I forgot to update the file. Give me a minute

Should be fixed now

File: piserver/app.py [hunk 1]
```diff
@@ -52,14 +52,14 @@ def handle_schedule():
...
-def post_command():
-    # TODO parse command, likely extract to new function once we are more clear 
-    # on the schema
-    position = request.form[ 'position' ]
-    duration = request.form[ 'time' ]
+@app.route( COMMAND_ROUTE, methods=[ 'POST', 'DELETE' ] )
+def handle_command():
+    if request.method == 'POST':
+        command = request.json
+        return smart_blinds_system.postBlindsCommand( command )
```
Comment (ishchow) [inline piserver/app.py]: postBlindsCommand takes in a dict, but then you pass it a json? Is that okay?

Comment (AY-Chen) [reply piserver/app.py]: `request.json` will convert the request data, provided it is a json, into a dict or something equivalent that can be accessed in the same way. I tested this with curl on localhost and it behaves as expected.

Comment (ishchow) [reply piserver/app.py]: okay, sounds good

File: blinds/blinds_command.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    is 0), or at the end of the day if duration is 0 or longer than the time remaining in the day    
+    '''
+    def toTimeBlock( self, currentTimeProvider=None ): 
+        if currentTimeProvider is None:
+            currentTime = datetime.datetime.utcnow().time()
+        else:
+            currentTime = currentTimeProvider()
+        startTime = datetime.time( currentTime.hour, currentTime.minute )
+
+        if ( self._duration == 0 ):
```
[the remark this answers is not in the document]
Comment (AY-Chen) [reply blinds/blinds_command.py]: So the choices are to make 0 and invalid value, or to leave it as a special value. I'm envisioning @kdehaan disallowing 0 as a value, but giving some kind of checkbox or something to allow the command to last until the end of the day, and then sending a value of 0 to the server. What do you think? Would you rather have it be an invalid value? If that's the case, how do we indicate to set it to the end of the day? It feels kind of weird to me to calculate a precise value on the app side and send that, or leave it on the user to do it (or the user to set a ridiculously large value to go over the end of the day)

[the remark this answers is not in the document]
Comment (ishchow) [reply blinds/blinds_command.py]: That's reasonable. I think using 0 is a better way than using either a negative value or a arbitrarily large positive value.

[the remark this answers is not in the document]
Comment (kdehaan) [reply blinds/blinds_command.py]: tbh 'end of day' scheduling hadn't occurred to me. I agree that 0 is better than negative or arbitrarily large, but I would have expected 0 to be reserved for a 'clear' command.

**Comment by AY-Chen** [comment]
> 
> 
> I'm not convinced that 0 as end-of-day is the best, but it should be fine for now. Overall it looks good to me, with the caveat that the exact API calls will likely have to change slightly when we have a better understanding of what we need.

I have a delete handler for the  clear request. I didn't feel like sending all the other required parts of a command(time, mode, etc) would be necessary if you only wanted to clear it.

**Pull request merged**